sensor_board/common/pinav_parser.c

```diff
@@ -0,0 +1,266 @@
+#include "pinav_parser.h"
+#include <string.h>
+#include <stdio.h>	// @TODO: remove when not needed for debugging
+
+typedef enum generic_status { FAIL = 0, SUCCESS = 1 } generic_status_t;
+
+// Private helper function to perform initial input validity checks
+static pinav_parser_status_t verify_inputs(pinav_parse_output_t * out, uint8_t * sentence) {
+	// Verify pointers
+	if (!out) {
+		return NULL_OUTPUT_PTR;
+	}
+	if (!sentence) {
+		out->id = NONE;
+		return NULL_SENTENCE_PTR;
+	}
+
+	// Verify that the sentence starts with '$'
+	if (*sentence != '$') {
+		out->id = NONE;
+		return SENTENCE_FORMAT_ERROR;
+	}
+
+	// Verify that the CR+LF sequence is within the 
+	// 80 charater limit specified by the datasheet
+	// @TODO Verify whether the CR+LF is included in the 80 char total
+	uint8_t * it;
+	uint8_t found = 0;
+	for (it = sentence; (it < sentence + 79) && (*it != '\0'); ++it) {
+		if ((*it == '\r') && (*(it + 1) == '\n')) {
+			found = 1;
+			break;
+		}
+	}
+	if (!found) {
+		out->id = NONE;
+		return IMPROPER_SENTENCE_LENGTH;
+	}
+
+	return OK;
+}
+
+// Private helper function to determine whether a string can be converted to a number
+static generic_status_t can_convert_stoi(const uint8_t * s, size_t len) {
+	generic_status_t result = SUCCESS;
+	
+	uint8_t * it;
+	for (it = s; it < s + len; ++it) {
+		if ((*it < '0') || (*it > '9')) {
+			result = FAIL;
+			break;
+		}
+	}
+
+	return result;
+}
+
+// Private helper function for converting string to int
+static generic_status_t stoi(uint32_t * i, uint8_t * s, size_t length) {
+	if (can_convert_stoi(s, length)) {
+		*i = *s - '0';
+		uint8_t * it;
+		for (it = s + 1; it < s + length; ++it) {
+			*i *= 10;
+			*i += *it - '0';
+		}
```

Comment:
Does not handle integer overflow. We may not need to handle it, but it's worth thinking about.